### PR TITLE
Item template expansion - Table Valued functions

### PR DESCRIPTION
**Pull request by ssreerama**
# Pull Request Template – vscode-mssql

## Description

This Pr adds:
- Table values functions template expansion for 'Add items...'
- Scenarios:
  - When the user creates a Table‑Valued Function, check whether the Functions folder exists in the project.
  - Determine the schema in which the user is creating the function.
  - If no schema is specified, default the schema to dbo.
  - If the Functions folder exists, place the new function file under the corresponding schema folder.
  - If the Functions folder does not exist, create the function file at the schema folder/project root level.
- Added tests to validate the scenarios
- Added test to validate the template loading

## Code Changes Checklist

-   [ ] New or updated **unit tests** added
-   [ ] All existing tests pass (`npm run test`)
-   [ ] Code follows [contributing guidelines](https://github.com/microsoft/vscode-mssql/blob/main/CONTRIBUTING.md)
-   [ ] Telemetry/logging updated if relevant
-   [ ] No regressions or UX breakage

## Reviewers: [Please read our reviewer guidelines](https://github.com/microsoft/vscode-mssql/blob/main/.github/REVIEW_GUIDELINES.md)

Tab_val_function_1

### CODE REVIEW

File: extensions/sql-database-projects/resources/templates/newTsqlTableValuedFunctionTemplate.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CREATE FUNCTION [@@SCHEMA_NAME@@].[@@OBJECT_NAME@@]
```
Comment (lewis-sanchez) [inline extensions/sql-database-projects/resources/templates/newTsqlTableValuedFunctionTemplate.sql:1]: Just curious, but why do we need a template?

Comment (ssreerama) [reply extensions/sql-database-projects/resources/templates/newTsqlTableValuedFunctionTemplate.sql:1]: Template files allow us to add consistent, predefined SQL object stubs and update or expand templates over time without code changes, while keeping object placement and rules in code.

File: extensions/sql-database-projects/test/projectController.test.ts [hunk 2]
```diff
@@ -221,15 +226,82 @@ suite('ProjectsController', function (): void {
...
+
+				// Test simple name (no schema)
+				result = projController.parseSchemaAndObjectName('MyFunction');
+				expect(result.schemaName, 'Default schema should be dbo').to.equal('dbo');
+				expect(result.objectName, 'Object name should remain unchanged').to.equal('MyFunction');
+
+				// Test edge case: leading dot
+				result = projController.parseSchemaAndObjectName('.MyFunction');
+				expect(result.schemaName, 'Leading dot should not be treated as schema').to.equal('dbo');
+				expect(result.objectName, 'Full string should be the object name').to.equal('.MyFunction');
```
Comment (lewis-sanchez) [inline extensions/sql-database-projects/test/projectController.test.ts:299]: The dot becomes part of the object name?

Comment (ssreerama) [reply extensions/sql-database-projects/test/projectController.test.ts:299]: yes, we do allow special characters in the name
